client/my-sites/domains/domain-management/transfer/transfer-to-other-user/index.jsx

```diff
@@ -325,7 +324,7 @@ class TransferOtherUser extends React.Component {
 	filterAvailableUsers( users ) {
 		return users.filter(
 			( user ) =>
-				! this.getWpcomUserId( user ) || this.getWpcomUserId( user ) !== this.props.currentUser.ID
+				getWpcomUserId( user ) !== false && getWpcomUserId( user ) !== this.props.currentUserId
```

Comment:
👍  This will now correctly exclude those Jetpack users without a matching WP.com account.